File: 1-Two-Sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
        array = [...nums]
    nums.sort((a, b) => a - b);
    let i = 0;
    let j = nums.length - 1;
    while (i <= j){
        if (nums[i] + nums[j] == target){
            break;
        }
        else if(nums[i] + nums[j] > target){
            j--;
        }
        else if(nums[i] + nums[j] < target){
            i++;
        }
    }
    const res = []
    for (x = 0; x <= array.length; x++){
        if (nums[i] == array[x]) {
           res.push(x)
           continue
        }
        if (nums[j] == array[x]) {
           res.push(x)
        }
    }
    return res
};

File: 1-Two-Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        sortnum = sorted(nums)
        index = []
        print(sortnum)
        left = 0
        right = len(sortnum) - 1
        while left < right:
            if sortnum[left] + sortnum[right] == target:
                break
            if sortnum[left] + sortnum[right] < target:
                left = left + 1
            if sortnum[left]+sortnum[right] > target:
                right = right - 1

        print(sortnum[left], sortnum[right])
        for i in range(0, len(nums)):
            if nums[i] == sortnum[left] or nums[i] == sortnum[right]:
                index.append(i)
        return index
        